src/gui/src/wg_navigationbar.cpp

```diff
@@ -76,9 +76,15 @@ unsigned int CNavigationBar::AddItem(SNavBarItem NavBarItem) {
 	m_Items.push_back(NavBarItem);
 	m_RenderedStrings.push_back(CRenderedString(m_pFontEngine, NavBarItem.sItemText, CRenderedString::VALIGN_BOTTOM, CRenderedString::HALIGN_CENTER));
     if (NavBarItem.sPictureFilename != "") {
-        m_Bitmaps.push_back(new CBitmapFileResourceHandle(NavBarItem.sPictureFilename));
-        // Set transparency color to COLOR_WHITE:
-        SDL_SetColorKey(m_Bitmaps.at(m_Bitmaps.size() - 1)->Bitmap(), SDL_SRCCOLORKEY, COLOR_WHITE.SDLColor(m_pSDLSurface->format));
+       try {
```

Comment:
That was my first idea too to fix this but in fact, I think it's better to keep it like this and to instead change the place where the exception is catched.
The reason is that any other exception in the code called by DeliverMessage would cause the same infinite loop.